packages/components/src/components/TokenInput/TokenInput.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+  FormControl,
+  FormLabel,
+  Input,
+  InputGroup,
+  InputProps,
+  InputRightElement,
+  Select,
+  SelectFieldProps,
+} from '@chakra-ui/react';
+import React from 'react';
+import { defaultTokens } from './tokens';
+
+export interface Token {
+  // token sybmol
+  symbol: string;
+  // token contract address
+  address: string;
+}
+
+export interface TokenInputProps {
+  /**
+   * an array of tokens to display
+   */
+  tokens?: Token[];
+  /**
+   * value of the input
+   */
+  value: string;
+  /**
```

Comment:
What's the difference between `value` and `selectedToken`?

Reply:
- `selectedToken` will contain the address of the selectedToken
- `value` will be the value of the input element

Reply:
@Nazeeh21 I think `amount` is a better name for the prop than `value`

Reply:
You should also rename `onValueChange` to `onAmountChange`

Reply:
Okay, I'll do that